consensus/state.go

```diff
@@ -1427,7 +1427,7 @@ func (cs *State) finalizeCommit(height int64) {
 	}
 
 	// Prune the blockStore.
-	if cs.config.RetainBlocks > 0 && uint64(cs.blockStore.Height()) > cs.config.RetainBlocks {
+	if cs.config.RetainBlocks > 0 && cs.blockStore.Height()-cs.blockStore.Base()+1 > int64(cs.config.RetainBlocks) {
```

Comment:
maybe it will be better to extract cs.blockStore.Height()-cs.blockStore.Base()+1 into a function
```go
(bs *BlockStore) NumRetainedBlocks() int { return bs.Height()-bs.Base()+1 }
```

Reply:
Good idea